src/Libraries/CoreNodes/GeometryColor.cs

```diff
@@ -60,25 +60,65 @@ public void Tessellate(IRenderPackage package, double tol = -1, int maxGridLines
             var surf = geometry as Surface;
             if (surf != null)
             {
+                var start = package.LineStripVertexColors.Count;
                 surf.PerimeterCurves().ForEach(
                         e =>
                             e.Tessellate(
                                 package,
                                 tol,
                                 maxGridLines));
+                var end = package.LineStripVertexColors.Count - 1;
+                ReColorVerticesFromTo(start, end, package);
             }
 
             var solid = geometry as Solid;
             if (solid != null)
             {
+                var start = package.LineStripVertexColors.Count;
                 solid.Edges.ForEach(
                         e =>
                             e.CurveGeometry.Tessellate(
                                 package,
                                 tol,
                                 maxGridLines));
+                var end = package.LineStripVertexColors.Count - 1;
+                ReColorVerticesFromTo(start, end, package);
             }
-    }
+
+            for (var i = 0; i < package.TriangleVertices.Count; i += 3)
+            {
+                NudgeVertexAlongVector(package.TriangleVertices, package.TriangleNormals, i, 0.001);
+            }
+
+        }
+
+        private void NudgeVertexAlongVector(IList<double> vertices, IList<double> normals, int i, double amount)
```

Comment:
If ever there was a need for shaders, this would be it.

I can't think of any better solution to this. Some ideads were:

* Make the default gray color a 90% alpha, but GeometryColor 100% alpha. Then the color would show through regardless of z fighting
* Automatically disable preview when connecting new nodes. This could be a feature you could enable or disable in a menu